docs/Compiler-Hardening-Guides/Compiler-Options-Hardening-Guide-for-C-and-C++.md

```diff
@@ -378,6 +392,27 @@ The performance overhead is dependent on the number of function’s instrumented
 
 ---
 
+### Implement control flow integrity checks
+
+| Compiler Flag                                                                                            | Supported since  | Description                                                  |
+|:-------------------------------------------------------------------------------------------------------- |:-------------:|:------------------------------------------------------------ |
+| <span id="-fcf-protection=full">`-fcf-protection=full`</span><br/>                                   |     GCC 8<br/>Clang 7.0.0          | Enable branch protection to counter Return Oriented Programming (ROP) and Jump Oriented Programming (JOP) attacks on many x86 architectures |
+| <span id="-mbranch-protection-standard">`-mbranch-protection=standard`</span>                        |     GCC 9<br/>Clang 8              | Enable branch protection to counter Return Oriented Programming (ROP) and Jump Oriented Programming (JOP) attacks on AArch64 |
+
+#### Synopsis
+
+Return-oriented programming (ROP) uses an initial subversion (such as a buffer overflow) to perform an indirect jump that executes an arbitrary sequence of instructions. A countermeasure is to ensure that jump addresses and return addresses are correct. This is not a complete solution, but it makes attacks harder to perform.
```

Comment:
I feel that this paragraph should mention "code reuse attacks". That makes the type of mitigation more clear.

When Arm introduced -mbranch-protection, they used this language in https://community.arm.com/arm-community-blogs/b/tools-software-ides-blog/posts/code-reuse-attacks-the-compiler-story

I could write this text if you like. I recently summarized similar text for https://bugs.launchpad.net/ubuntu/+source/gcc-13/+bug/2040321

Reply:
I'll add some text, hopefully you'll like it.